RallyHereDebugTool/Source/Private/RHDTW_Automation.cpp

```diff
@@ -38,10 +38,14 @@ void FRHDTW_Automation::Init(URallyHereDebugTool* InOwner, const FString& InName
 	AutomationController = AutomationControllerModule->GetAutomationController();
 
 	AutomationController->Init();
-	AutomationController->OnTestsRefreshed().AddSPLambda(this, [this]()
+	AutomationController->OnTestsRefreshed().AddLambda([this, WeakThis=AsWeak()]()
 		{
-			// Reset the filter string when the tests are refreshed
-			bHasInitializedFilter = false;
+			// use weak pointer to validate this pointer is still valid (this is a workaround for some engine versions not having CreateLambdaSP())
+			if (WeakThis.IsValid())
```

Comment:
Unfortunately, just checking if the weak pointer is valid is not technically enough to ensure that this doesn't get cleaned up.  It has to create a reference to the shared pointer in this scope - to ensure that it is not cleaned up in the middle of this operation.
```
if (auto sharedThis = WeakThis.Pin())
{
    ...
}
```

Reply:
I had that originally, but the pinned pointer doesnt have access to the internals of the class without a "this" capture.  I can probably pin it for safety then use the this pointer I guess?

I generally dont like this approach and the LambdaSP method was preferred, but it was added in 5.3 =/